### DISCUSSION TITLE
Saleor Core V3.0 Docker installation

### DISCUSSION THREAD

**Discussion by laportem** [saleor/saleor#8046] [Deployment]
Has anyone managed to installed saleor core v3.0 using docker-compose. 
Can you please share your docker-compose?

I’ve git clone the version 3 branch as follows:
`git clone -b 3.0 --single-branch https://github.com/mirumee/saleor.git api`

And the snipet to build the core app my docker-compose file is as follows
`
  api:
    ports:
      - 8000:8000
    build:
      context: ./api
      dockerfile: ./Dockerfile
      args:
        STATIC_URL: "/static/"
    restart: unless-stopped
    networks:
      - saleor-network
    depends_on:
      - db
      - redis
      - jaeger
    volumes:
      - ./data/saleor/:/app/saleor:Z
      - ./data/templates/:/app/templates:Z
      - ./data/tests/:/app/tests
      # shared volume between worker and api for media
      - saleor-media:/app/media
    command: python manage.py runserver 192.168.33.12:8000
    env_file: common.env
    environment:
      - JAEGER_AGENT_HOST=jaeger
      - STOREFRONT_URL=http://192.168.33.12:3000/
      - DASHBOARD_URL=http://192.168.33.12:9000/
`

But I keep getting this error on docker-compose up

**ModuleNotFoundError: No module named 'saleor.settings'
ModuleNotFoundError: No module named 'saleor.celeryconf'**

Any idea where I am going wrong.

**Comment by karolkielecki** [comment]
Hi! Maybe you can find a solution under this thread: https://github.com/mirumee/saleor/issues/4323. If the issue still appears, please let me know. Cheers

**Comment by koko-ng** [comment]
Hi!
I had the same problem, for me the solution was to change the volumes of the API and the worker from Z to z as they are shared between the two containers (More details here: [https://stackoverflow.com/a/35222815/5728258)](https://stackoverflow.com/a/35222815/5728258). I also had a similar problem with the storefront and dashboard containers where I had to add the flag Z to the /app volumes.
Note that I am using podman and podman-compose instead of Docker, the problem might also arise from that, I didn't had the time to test with docker.

Full diff:
```diff
diff --git a/docker-compose.yml b/docker-compose.yml
index bbd77c4..70244a1 100644
--- a/docker-compose.yml
+++ b/docker-compose.yml
@@ -17,8 +17,8 @@ services:
       - redis
       - jaeger
     volumes:
-      - ./saleor/saleor/:/app/saleor:Z
-      - ./saleor/templates/:/app/templates:Z
+      - ./saleor/saleor/:/app/saleor:z
+      - ./saleor/templates/:/app/templates:z
       - ./saleor/tests/:/app/tests
       # shared volume between worker and api for media
       - saleor-media:/app/media
@@ -44,7 +44,7 @@ services:
     depends_on:
       - api
     volumes:
-      - ./saleor-storefront/:/app:cached
+      - ./saleor-storefront/:/app:Z,cached
       - /app/node_modules/
     # Nginx is used to proxy SSR requests thru docker networking
     command: sh -c '(nginx &) && npm run start'
@@ -57,7 +57,7 @@ services:
       - 9000:9000
     restart: unless-stopped
     volumes:
-      - ./saleor-dashboard/:/app:cached
+      - ./saleor-dashboard/:/app:Z,cached
       - /app/node_modules/
     command: npm start -- --host 0.0.0.0
 
@@ -99,8 +99,8 @@ services:
       - redis
       - mailhog
     volumes:
-      - ./saleor/saleor/:/app/saleor:Z,cached
-      - ./saleor/templates/:/app/templates:Z,cached
+      - ./saleor/saleor/:/app/saleor:z,cached
+      - ./saleor/templates/:/app/templates:z,cached
       # shared volume between worker and api for media
       - saleor-media:/app/media
```

Hope it helps :)
